### DISCUSSION TITLE
How to get `source` of ROOT?

### DISCUSSION THREAD

**Discussion by xxlv** [graphql-java/graphql-java#2860] [Q&A] [answered]
Hello, developers,

when i use query like 

```
query{
    user{
     friends{
        name 
   }
}
}
```
when fetching field `name`,  i want to get the `user` field's SOURCE. but now i can only read `friends`'s  **SOURCE**

am i wrong?

i think, a query should contains all SOURCE chain from parent field to child field.

Example: 

when feching `friends.name`. i want to read  field `user's` result as SOURCE

**Comment by bbakerman** [comment, marked as answer]
The root source object needs to be setup before you enter the engine.

`graphql.ExecutionInput.Builder#root`

If you dont do this there is no "root" source.  Often the root field data fetchers dont care for a source but if you do - this is how you set it

**Reply by xxlv** [reply]
Thank you for your help

I'm work on dyncmic configable graphql.. so in my case, i need SET root source and make sure every field under ROOT field could see the  ROOT source 

when root field fetched value ,i store the data as attached context for root field . it works now .
